app/models/service_template.rb

```diff
@@ -23,6 +23,19 @@ class ServiceTemplate < ApplicationRecord
     "redhat"                => _("RHEV"),
     "vmware"                => _("VMware")
   }
+  PROV_TYPE_TO_MODEL = {
+    "amazon"                   => :ServiceTemplate,
+    "azure"                    => :ServiceTemplate,
+    "generic"                  => :ServiceTemplate,
+    "generic_orchestration"    => :ServiceTemplateOrchestration,
+    "generic_ansible_tower"    => :ServiceTemplateAnsibleTower,
+    "generic_ansible_playbook" => :ServiceTemplateAnsiblePlaybook,
+    "google"                   => :ServiceTemplate,
+    "microsoft"                => :ServiceTemplate,
+    "openstack"                => :ServiceTemplate,
+    "redhat"                   => :ServiceTemplate,
+    "vmware"                   => :ServiceTemplate
+  }.freeze
```

Comment:
wouldn't logic like the following work here ? 

```
return klass if request_type == ServiceTemplate.name

request_klass = ServiceTemplate.descendants.detect { |sub_klass| request_type == sub_klass.name }
return request_klass if request_klass
```

since: 
```
ServiceTemplate.descendants.collect(&:name)
=> ["ServiceTemplateAnsibleTower", "ServiceTemplateGeneric", "ServiceTemplateOrchestration", "ServiceTemplateAnsiblePlaybook"]
```